tests/ut_common_copyfilessubdir.py

```diff
@@ -0,0 +1,71 @@
+""" 
+ut_common_copyfilessubdir.py --- test if build_component.py can copy files to subdirectories 
+of the build target dir.
+""" 
+
+import os
+import sys 
+import subprocess
+
+def main():
+    # set component directory
+    component_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+    # change directory to component directory ../../common/
+    os.chdir(component_dir)
+
+    #subprocess.call(["python", "initialize.py"])
+    #subprocess.call(["python","scripts/build.py","-t"])
+    sub = subprocess.Popen([sys.executable, 'scripts/build.py', '-t'], 
+                                         stderr=subprocess.PIPE, 
+                                         stdout=subprocess.PIPE)
+    (out, err) = sub.communicate()
+
+    if err != '':
+        print "FAIL"
+
+    if not "Done building!" in out:
+        print "FAIL"
+
+    # open config_build.py and read each lines
```

Comment:
I suggest to refactor this to
* Build a dummy component specific to this unit test,
* which has a separate build config,
* and instead of re-reading the build config, compare the resulting file list after building with a hardcoded file list.

This avoids testing the correctness of a function (reading in the build config in this case) using the same function (i.e. the code below this comment, copied in from build_component.py)